meteor/lib/api/rest.ts

```diff
@@ -0,0 +1,204 @@
+	 * @param rundownPlaylistId Playlist to activate.
+	 * @param rehearsal Whether to activate into rehearsal mode.
+	 * @param connection Connection data including client and header details
+	 */
+	activate(
+		rundownPlaylistId: RundownPlaylistId,
+		rehearsal: boolean,
+		connection: Meteor.Connection
+	): Promise<ClientAPI.ClientResponse<void>>
+	/**
+	 * Deactivates a Playlist.
+	 *
+	 * Throws if the Playlist is not currently active.
+	 * @param rundownPlaylistId Playlist to deactivate.
+	 * @param connection Connection data including client and header details
+	 */
+	deactivate(
+		rundownPlaylistId: RundownPlaylistId,
+		connection: Meteor.Connection
+	): Promise<ClientAPI.ClientResponse<void>>
+	/**
+	 * Executes the requested action by passing `actionId` directly to the blueprints `executeAction` function.
+	 *
+	 * Throws if the target Playlist is not currently active.
+	 * Throws if there is not an on-air Part.
+	 * Throws if the `executeAction` method of the blueprints throws - blueprint error will be returned to caller.
+	 * @returns An object in the form {queuedPartInstanceId?: PartInstanceId, taken?: boolean} (if a part was queued by the action/if the next part was automatically taken)
+	 * @param rundownPlaylistId Playlist to execute action in.
+	 * @param actionId Action Id string, should match something inside the blueprints.
+	 * @param userData Any value, recommended to be a relatively simple { [key: string]: string }-style object, as the object may need to survive a round of serialization.
+	 * @param connection Connection data including client and header details
+	 */
+	executeAction(
+		rundownPlaylistId: RundownPlaylistId,
+		actionId: string,
+		userData: any,
+		connection: Meteor.Connection
+	): Promise<ClientAPI.ClientResponse<object>>
+	/**
+	 * Executes the requested AdLib/AdLib Action. This is a "planned" AdLib (Action) that has been produced by the blueprints during the ingest process.
+	 *
+	 * Throws if the target Playlist is not active.
+	 * Throws if there is not an on-at part instance.
+	 * @returns a `ClientResponseError` if an adLib for the provided `adLibId` cannot be found.
+	 * @param rundownPlaylistId Playlist to execute adLib in.
+	 * @param adLibId AdLib to execute.
+	 * @param connection Connection data including client and header details
+	 * @param triggerMode A string to specify a particular variation for the AdLibAction, valid actionType strings are to be read from the status API.
+	 */
+	executeAdLib(
+		rundownPlaylistId: RundownPlaylistId,
+		adLibId: AdLibActionId | RundownBaselineAdLibActionId | PieceId | BucketAdLibId,
+		connection: Meteor.Connection,
+		triggerMode?: string
+	): Promise<ClientAPI.ClientResponse<object>>
+	/**
+	 * Moves the next point by `delta` places. Negative values are allowed to move "backwards" in the script.
+	 *
+	 * Throws if the target Playlist is not active.
+	 * Throws if there is not next Part set (e.g. rundown is empty)
```

Comment:
There can be no next Part set, if this is the end of the Rundown, I suppose it shouldn't throw then, right?

Reply:
I've clarified the wording, I think it's ok to say that this throws if a next Part couldn't be found because in that case we have failed to perform the requested action/the requested action is not valid? https://github.com/evs-broadcast/sofie-core/blob/66d52dc36c0b2e4b0bcf768261fd35fe9218aa9b/packages/openapi/api/definitions/playlists.yaml#L146

Reply:
Julian made a similar point, my reply here https://github.com/nrkno/sofie-core/pull/777#discussion_r1085159421